pkg/controllers/clusteroperator_controller.go

```
@@ -192,6 +204,76 @@ func (r *CloudOperatorReconciler) applyResources(ctx context.Context, resources
+	for _, resource := range resources {
+		switch resource.(type) {
+		case *appsv1.Deployment:
+			live := &appsv1.Deployment{}
+			if err := r.Get(ctx, client.ObjectKeyFromObject(resource), live); apierrors.IsNotFound(err) {
+				klog.V(2).Infof("Deployment %s not found, treating as not converged", resource.GetName())
+				return false, nil
+			} else if err != nil {
+				return false, fmt.Errorf("failed to get Deployment %s: %w", resource.GetName(), err)
+			}
+			if isDeploymentRolloutStalled(live) {
+				return false, fmt.Errorf("Deployment %s rollout stalled: ProgressDeadlineExceeded", resource.GetName())
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not be capitalized

(staticcheck)